File: android/src/main/kotlin/com/braze/brazeplugin/BrazeUIHandler.kt
package com.braze.brazeplugin

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

/**
 * Handles UI events for native Android views.
 */
class BrazeUIHandler(binaryMessenger: BinaryMessenger) : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null

    init {
        EventChannel(binaryMessenger, "braze_banner_view_channel").setStreamHandler(this)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    fun sendResizeEvent(height: Double, identifier: String) {
        val resizeData = mapOf("height" to height, "containerId" to identifier)
        eventSink?.success(resizeData)
    }
}

File: ios/braze_plugin/Sources/braze_plugin/BrazeUIHandler.swift
import Flutter

/// Handles UI events for native iOS views.
public class BrazeUIHandler : NSObject, FlutterStreamHandler {
  private var eventSink: FlutterEventSink?

  init(messenger: FlutterBinaryMessenger) {
    super.init()
    let channel = FlutterEventChannel(name: "braze_banner_view_channel", binaryMessenger: messenger)
    channel.setStreamHandler(self)
  }

  public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
    eventSink = events
    return nil
  }

  public func onCancel(withArguments arguments: Any?) -> FlutterError? {
    eventSink = nil
    return nil
  }

  func sendResizeEvent(height: Double, identifier: String) {
    let resizeData: [String: Any] = ["height": height, "containerId": identifier]
    eventSink?(resizeData)
  }

  func sendDismissEvent(placementId: String, stableKey: String, trackingId: String) {
    let dismissData: [String: Any] = [
      "action": "dismiss",
      "placementId": placementId,
      "stableKey": stableKey,
      "trackingId": trackingId,
    ]
    eventSink?(dismissData)
  }
}
